src/common/utils/registry/auth/apikey.go

```diff
@@ -0,0 +1,38 @@
+package auth
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/goharbor/harbor/src/common/http/modifier"
+)
+
+type apiKeyAuthorizer struct {
+	key   string
+	value string
+	in    string
+}
+
+// NewAPIKeyAuthorizer returns a apikey authorizer
+func NewAPIKeyAuthorizer(key, value, in string) modifier.Modifier {
```

Comment:
The standard token flow should work for Quay, right? Or the docker client cannot work with it. So why we need this APIKeyAuthorizer? How can I get the API key from Quay.io and is there any document about how to use the API key?

Reply:
Just want to add a more flexible authorizer which can customize location in header or query,  and auth content. Because some tool maybe use not standard auth way like private(access) token etc... , if we dont want to add a new more authorizer, i can remove it temporary.

Reply:
How can I get the API key from Quay.io and is there any document about how to use the API key?

Reply:
It's oauth2, https://docs.quay.io/api/

Reply:
Sorry for the late reply. Can users use the username/password to do the authentication?